### PR TITLE
Expiration of Contracts and Bids

### PR DESCRIPTION
**Pull request by ljmcgann**
Expanded functionality for the manager service to expire bids and
contracts. Expands database api and business objects as needed to
accomplish this. TG-156

### CODE REVIEW

**Comment by ljmcgann** [comment]
This code can't be finished until Ali finishes the contracts business objects

File: flocx_market/manager/service.py [hunk 1]
```diff
@@ -43,8 +45,24 @@ def update_expired_offers(self, context):
...
         unexpired_offers = Offer.get_all_unexpired()
+        exp = 0
         for offer in unexpired_offers:
             if offer.end_time < now:
                 offer.expire()
-        if len(unexpired_offers) > 0:
-            LOG.info("Updated " + str(len(unexpired_offers)) + " offers")
+                exp += 1
+        if exp > 0:
+            LOG.info("Updated " + str(exp) + " offers")
```
Comment (tzumainn) [inline flocx_market/manager/service.py]: Nice catch!

**Comment by tzumainn** [comment]
Looks good! The contract business object layer stuff just got merged, so I believe this can be completed now.

File: flocx_market/objects/bid.py [hunk 2]
```diff
@@ -51,3 +51,13 @@ def save(self):
         db_bid = db.bid_update(
             self.marketplace_bid_id, updates)
         return self._from_db_object(self, db_bid)
+
+    @classmethod
+    def get_all_unexpired(cls):
+        unexpired = db.bid_get_all_unexpired()
+        print(type(unexpired[0]))
```
Comment (tzumainn) [inline flocx_market/objects/bid.py]: I'm guessing this was left in by mistake?

Comment (ljmcgann) [reply flocx_market/objects/bid.py]: yeah

**Pull request merged**